packages/react-keyring/test/Authenticator.spec.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react'
+import { test, expect, vi, beforeEach, afterEach } from 'vitest'
+import 'fake-indexeddb/auto'
+import userEvent from '@testing-library/user-event'
+import { render, screen } from '@testing-library/react' // (or /dom, /vue, ...)
+import { unmountComponentAtNode } from 'react-dom'
+import { act } from 'react-dom/test-utils'
+
+import { KeyringContext, keyringContextDefaultValue, KeyringContextValue } from '../src/providers/Keyring'
+import { Authenticator, CancelButton, Form, EmailInput } from '../src/index'
+
+let container: HTMLDivElement = document.createElement('div')
+beforeEach(() => {
+  // setup a DOM element as a render target
+  container = document.createElement('div')
+  document.body.appendChild(container)
+})
+
+afterEach(() => {
+  // cleanup on exiting
+  unmountComponentAtNode(container)
+  container.remove()
+})
+
+test('CancelButton', async () => {
+  const user = userEvent.setup()
+
+  const cancelRegisterSpace = vi.fn()
+  const contextValue: KeyringContextValue = [
+    keyringContextDefaultValue[0],
+    { ...keyringContextDefaultValue[1], cancelRegisterSpace }
+  ]
+  act(() => {
+    render((
```

Comment:
I'm new to testing-library, but i like the look of it and its goals. From looking at some examples it looks like we should be able to avoid the beforeEach / afterEach boilerplate and maybe the `act` step can go too? If we can get our tests looking more like this example that'd be rad https://github.com/kentcdodds/react-testing-library-course/blob/5abc1039d677767b8de5197c049bfd0428fa7420/src/__tests__/app-03.js#L12-L38

Reply:
a handy list of things to avoid https://kentcdodds.com/blog/common-mistakes-with-react-testing-library which can be used to extrapolate what should be possible